pmd-java/src/test/resources/net/sourceforge/pmd/lang/java/rule/bestpractices/xml/MethodReturnsInternalArray.xml

```diff
@@ -266,4 +266,116 @@ public enum MethodReturnsInternalArrayCaseEnum {
         ]]></code>
     </test-code>
 
+    <test-code>
+        <description>Inner interface</description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+public interface OuterInterface {
+       interface InnerInterface {
+            String method();
+       }
+}
+        ]]></code>
+    </test-code>
+
+    <test-code>
+        <description>Inner annotation</description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Retention;
+
+public final class I {
+    private I(){
+    }
+    @Retention(RetentionPolicy.RUNTIME)
+    public static @interface Inner {}
+}
+        ]]>
+        </code>
+    </test-code>
+    
+    <test-code>
+        <description>Top-level annotation</description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Target(ElementType.METHOD)
+@Retention(RetentionPolicy.RUNTIME)
+public @interface TopLevelAnnotation {
+
+}
+        ]]></code>
+    </test-code>
+
+    <test-code>
+        <description>Enum that returns array instead of copy</description>
+        <expected-problems>1</expected-problems>
+        <code><![CDATA[
+public enum MethodReturnsInternalArrayCaseEnum {
+    ONE("One"),
+    TWO("Two", "Three");
+
+    private String[] titles;
+
+    MethodReturnsInternalArrayCaseEnum(String... titles) {
+        this.titles = titles;
+    }
+
+    public String[] getTitles() {
+        return titles;
+    }
+}
+        ]]></code>
+    </test-code>
+
+    <test-code>
+        <description>Inner enum that returns array instead of copy</description>
+        <expected-problems>1</expected-problems>
+        <code><![CDATA[
+public class OuterClass {
+
+    public enum InnerEnum {
+        INNER_ENUM("first", "second");
+        private String[] titles;
+
+        InnerEnum(String... titles) {
+            this.titles = titles;
+        }
+
+        public String[] getTitles() {
+            return titles;
+        }
+    }
+}
+        ]]></code>
+    </test-code>
+
+    <test-code>
+        <description>Inner enum that returns array instead of copy</description>
```

Comment:
This test description is misleading, it should be "Inner enum that returns copy of array".

Reply:
whoops, copy-pasting